packages/models/src/lib/reports-diff.ts

```diff
@@ -0,0 +1,139 @@
+import { type ZodTypeAny, z } from 'zod';
+import {
+  auditDisplayValueSchema,
+  auditOutputSchema,
+  auditValueSchema,
+} from './audit-output';
+import { commitSchema } from './commit';
+import {
+  executionMetaSchema,
+  packageVersionSchema,
+  scoreSchema,
+  slugSchema,
+  titleSchema,
+} from './implementation/schemas';
+import { pluginMetaSchema } from './plugin-config';
+
+function makeComparisonSchema<T extends ZodTypeAny>(schema: T) {
+  const sharedDescription = schema.description || 'Result';
+  return z.object({
+    before: schema.describe(`${sharedDescription} (source commit)`),
+    after: schema.describe(`${sharedDescription} (target commit)`),
+  });
+}
+
+function makeArraysComparisonSchema<
+  TDiff extends typeof scorableDiffSchema,
+  TResult extends ZodTypeAny,
```

Comment:
Question - do we lose any type safety by having this as `any` compared to how it was before?

Reply:
No, because it's a generic type param, the return type is inferred based on the actual parameter type. The `extends ZodTypeAny` is just an extra constraint to ensure the parameter type is some Zod schema.

Reply:
Thank you for clarifying.